extension/src/plots/model/collect.ts

```diff
@@ -67,256 +50,109 @@ import {
-import { typedValueTreeEntries } from '../../experiments/columns/collect/metricsAndParams'
-
-type CheckpointPlotAccumulator = {
-  iterations: Record<string, number>
-  plots: Map<string, CheckpointPlotValues>
-}
-
-const collectFromMetricsFile = (
-  acc: CheckpointPlotAccumulator,
-  name: string,
-  iteration: number,
-  key: string | undefined,
-  value: Value | ValueTree,
-  ancestors: string[] = []
-) => {
-  const pathArray = [...ancestors, key].filter(Boolean) as string[]
-
-  if (isValueTree(value)) {
-    for (const [childKey, childValue] of typedValueTreeEntries(value)) {
-      collectFromMetricsFile(
-        acc,
-        name,
-        iteration,
-        childKey,
-        childValue,
-        pathArray
-      )
-    }
-    return
-  }
-
-  const path = appendColumnToPath(...pathArray)
 
-  addToMapArray(acc.plots, path, { group: name, iteration, y: value })
-}
-
-type MetricsAndDetailsOrUndefined =
-  | {
-      checkpoint_parent: string | undefined
-      checkpoint_tip: string | undefined
-      metrics: MetricOrParamColumns | undefined
-      status: ExperimentStatus | undefined
-    }
-  | undefined
-
-const transformExperimentData = (
-  experimentFieldsOrError: ExperimentFieldsOrError
-): MetricsAndDetailsOrUndefined => {
-  const experimentFields = experimentFieldsOrError.data
-  if (!experimentFields) {
-    return
-  }
-
-  const { checkpoint_tip, checkpoint_parent, status } = experimentFields
-  const { metrics } = extractColumns(experimentFields)
-
-  return { checkpoint_parent, checkpoint_tip, metrics, status }
-}
-
-type ValidData = {
-  checkpoint_parent: string
-  checkpoint_tip: string
-  metrics: MetricOrParamColumns
-  status: ExperimentStatus
-}
-
-const isValid = (data: MetricsAndDetailsOrUndefined): data is ValidData =>
-  !!(data?.checkpoint_tip && data?.checkpoint_parent && data?.metrics)
+export const getCustomPlotId = (metric: string, param = CHECKPOINTS_PARAM) =>
+  `custom-${metric}-${param}`
 
-const collectFromMetrics = (
-  acc: CheckpointPlotAccumulator,
-  experimentName: string,
-  iteration: number,
-  metrics: MetricOrParamColumns
+const collectCheckpointValuesFromExperiment = (
+  values: CheckpointPlotValues,
+  exp: ExperimentWithDefinedCheckpoints,
+  splitUpMetricPath: string[]
 ) => {
-  for (const file of Object.keys(metrics)) {
-    collectFromMetricsFile(
-      acc,
-      experimentName,
-      iteration,
-      undefined,
-      metrics[file],
-      [file]
-    )
-  }
-}
-
-const getLastIteration = (
-  acc: CheckpointPlotAccumulator,
-  checkpointParent: string
-): number => acc.iterations[checkpointParent] || 0
-
-const collectIteration = (
-  acc: CheckpointPlotAccumulator,
-  sha: string,
-  checkpointParent: string
-): number => {
-  const iteration = getLastIteration(acc, checkpointParent) + 1
-  acc.iterations[sha] = iteration
-  return iteration
-}
+  const group = exp.name || exp.label
+  const maxEpoch = exp.checkpoints.length + 1
```

Comment:
I just realized that using this way to get iterations breaks modified experiments. I need to understand a bit more on how iterations are originally gathered and see if it's even it's a good idea to get iterations through experiments... 

If not, I either need to see if we can gather both plots through output data or rethink how to consolidate when plot plots require two different kinds of data.

Reply:
After a discussion with Matt, we decided to keep the current implementation for now, since with a future change in the exp show output, it's likely that the current way of collecting iterations will no longer work anyway.

Reply:
Asked here: https://github.com/iterative/dvc/pull/9170#issuecomment-1473001613

Reply:
Answer here: https://github.com/iterative/dvc/pull/9170#issuecomment-1473029067